### DISCUSSION TITLE
react hook form library throwing an error for Checkbox value.includes is not a function

### DISCUSSION THREAD

**Discussion by lakhandeshpande** [react-hook-form/react-hook-form#3325] [Q&A]
![image](https://user-images.githubusercontent.com/28001038/97656274-ed7ce280-1acb-11eb-95ab-34a1f30c5d24.png)

See attached image reference : 

I am setting the correct value for checkbox which is either true or false but the react hook form is not dealing with Boolean values and throwing this error. Could you help me out of this core library issue? Do I need to upgrade the react hook form? 

Currently I am using 6.0.5.

**Comment by bluebill1049** [comment]
hmmm any chance you can include a polyfill

**Reply by lakhandeshpande** [reply]
Where can I find it? Can you help?

**Reply by lakhandeshpande** [reply]
babel-polyfill@^6.26.0:
  version "6.26.0"
  resolved "https://registry.yarnpkg.com/babel-polyfill/-/babel-polyfill-6.26.0.tgz#379937abc67d7895970adc621f284cd966cf2153"
  dependencies:
    babel-runtime "^6.26.0"
    core-js "^2.5.0"
    regenerator-runtime "^0.10.5"

INSIDE yarn.lock

**Reply by lakhandeshpande** [reply]
@bluebill1049  I am bit novice to React-hook-form, just wanted to get rid of this issue  as its blocking me.

**Reply by bluebill1049** [reply]
hey, 

would like to include this script for now to move on with this IE 11 problem.

```ts
if (!String.prototype.includes) {
  String.prototype.includes = function(search, start) {
    'use strict';

    if (search instanceof RegExp) {
      throw TypeError('first argument must not be a RegExp');
    } 
    if (start === undefined) { start = 0; }
    return this.indexOf(search, start) !== -1;
  };
}
```

**Reply by lakhandeshpande** [reply]
@bluebill1049  Where should I use this snippet in?

**Reply by bluebill1049** [reply]
you can play at the entry of your app, index.js
it will work as a polyfill.

**Reply by lakhandeshpande** [reply]
So, Anywhere in the index.js I just need to push this snippet and it will work? How can I connect these two points together?

**Reply by bluebill1049** [reply]
no, just at your root.

**Reply by lakhandeshpande** [reply]
@bluebill1049 , I am not sure what I am missing here but it seems its not working for me....still getting the same error.
I've placed the snippet in my root.

I even tried with upgrading the version from 6.0.5 to 6.10.1 but still the same issue. Help!!

**Reply by bluebill1049** [reply]
not sure why it's not working...

```
import 'react-app-polyfill/ie11';
import 'react-app-polyfill/stable';
```

try those two packages at your `index.js`

**Reply by lakhandeshpande** [reply]
@bluebill1049  Do you mean to say, import these two in index.js and also have that snippet at my root or just either of them?

**Reply by lakhandeshpande** [reply]
I don't have that library already , do I need to install it first and then include it in the index.js?

**Reply by bluebill1049** [reply]
install them

**Reply by lakhandeshpande** [reply]
if (isCheckBoxInput(ref) && options) {
            options.length > 1
                ? options.forEach(function (_a) {
                    var checkboxRef = _a.ref;
                    return (checkboxRef.checked = (value.toString()).includes(checkboxRef.value));
                })
                : (options[0].ref.checked = !!value);
        }

@bluebill1049 

I've made a change in **node_modules\react-hook-form\dist\index.ie11.development.js** added **toString** inside the if and its working as expected, the value always needs to be a string that's what currently expecting. But this is not recommended, since my changes will be gone when I reinstall the module. What's the work around for this?

**Reply by lakhandeshpande** [reply]
NB: On the other hand, I am using mono repo and workspaces for my project just to give you an idea how am I currently working around with my project. Where exactly these two imports needs to be added at workspace root or target package?

**Comment by hornetDC** [comment]
I am getting same error with gatsby. It occurs when checkbox unmounts.
